File: src/main/java/ru/mdashlw/hypixel/entities/player/Stats.java
package ru.mdashlw.hypixel.entities.player;

import com.fasterxml.jackson.databind.node.ObjectNode;
import ru.mdashlw.hypixel.entities.player.stats.SkyWars;
import ru.mdashlw.hypixel.entities.player.stats.UHC;
import ru.mdashlw.hypixel.util.DynamicObjectNode;

/**
 * Player stats.
 */
public final class Stats extends DynamicObjectNode {
    Stats(final ObjectNode node) {
        super(node);
    }

    /**
     * Stats on SkyWars.
     *
     * @return SkyWars stats or null.
     */
    public SkyWars getSkyWars() {
        return this.get("SkyWars", null, node -> new SkyWars((ObjectNode) node));
    }

    /**
     * Stats on UHC Champions.
     *
     * @return UHC stats or null.
     */
    public UHC getUHC() {
        return this.get("UHC", null, node -> new UHC((ObjectNode) node));
    }
}

File: src/main/kotlin/ru/mdashlw/hypixel/api/entities/player/stats/Stats.kt
package ru.mdashlw.hypixel.api.entities.player.stats

import com.fasterxml.jackson.databind.node.ObjectNode
import ru.mdashlw.hypixel.api.entities.player.stats.games.SkyWars
import ru.mdashlw.hypixel.api.entities.player.stats.games.UHC
import ru.mdashlw.hypixel.api.util.CustomObjectNode
import ru.mdashlw.hypixel.api.util.get
import ru.mdashlw.hypixel.api.util.obj

@Suppress("PropertyName")
class Stats(obj: ObjectNode) : CustomObjectNode(obj) {
    val SkyWars: SkyWars?
        get() = get("SkyWars", null) { SkyWars(it.obj()) }

    val UHC: UHC?
        get() = get("UHC", null) { UHC(it.obj()) }
}
